examples/Containerfile

```diff
@@ -0,0 +1,40 @@
+# Example Containerfile for KFP Component Base Image
+#
+# This demonstrates best practices for creating custom base images
+# for Kubeflow Pipelines components.
+#
+# The resulting image will be pushed to:
+#   ghcr.io/kubeflow/pipelines-components-example:main
+#
+# Usage in a component:
+#   @dsl.component(base_image="ghcr.io/kubeflow/pipelines-components-example:main")
+#   def my_component():
+#       import pandas as pd
+#       ...
+
+FROM python:3.11-slim
+
+LABEL org.opencontainers.image.title="KFP Components Example Base Image"
+LABEL org.opencontainers.image.description="Example base image with common ML dependencies for Kubeflow Pipelines components"
+LABEL org.opencontainers.image.source="https://github.com/kubeflow/pipelines-components"
+LABEL org.opencontainers.image.licenses="Apache-2.0"
+LABEL maintainer="Kubeflow Pipelines"
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+
+RUN pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir \
+        numpy==1.26.4 \
+        pandas==2.2.0 \
+        scikit-learn==1.4.0
+
+WORKDIR /app
+
+# Create a non-root user and switch to it for running components
+RUN groupadd --system kfp && \
+    useradd --system --gid kfp --create-home --home-dir /home/kfp kfp && \
+    chown -R kfp:kfp /app
+
+USER kfp
```

Comment:
While good security practice to do this for standard k8s environments, it could inadvertently make the images not work on envirnoments like OpenShift which run the container with arbitrary ids. 

We can't enforce that behavior here but we can certainly add as a comment that this is: 

1. Optional
2. It provides added security, but it can make the image not work out of box in environments like Openshift
3. Provide a commented out command that also adds the directory permissions to [group 0 ](https://docs.redhat.com/en/documentation/openshift_container_platform/3.11/html/creating_images/creating-images-guidelines)so arbitrary users in Openshift can also run the app - and mention that other distributions might have different requirements so they should do their research.